### PR TITLE
VMO-3369 toggle view and edit

### PR DESCRIPTION
**Pull request by poojahpatil90**
VMO-3369

Builder canvas - Read-only / view-mode of designer canvas elements (+ hook up Edit Flow button without page reload)

### CODE REVIEW

File: src/components/common/AudioLibrarySearchField.vue
Comment (safydy-r) [inline src/components/common/AudioLibrarySearchField.vue:73]: @bzabos , can you remind us our rule for semi-colon ?
If we expect trailing semi, shall we add a rule in `eslintrc` to avoid this kind of change ?

File: src/components/common/AudioLibrarySearchField.vue
Comment (safydy-r) [inline src/components/common/AudioLibrarySearchField.vue:72]: @poojahpatil90 , thanks for thinking of these indentation etc, but I would avoid making auto fix like this before PR review. This is a big distraction for reviewer especially for big PR.

My suggestion is to wait for PR approval, then commit on last change to make auto fix from IDE before merge, what do you think?

File: src/components/common/AudioLibrarySearchField.vue
Comment (safydy-r) [inline src/components/common/AudioLibrarySearchField.vue:6]: This is good for a 1st PR in flow-builder.
I noticed you introduced `isEditdable` prop in many components to disable the side bar children.
I would suggest we take advantage of the `<fieldset>` like here:  https://bitbucket.org/voto/voto5/src/e642ee81827f72c654cced81c9e304e80e4d64ea/resources/assets/js/trees/components/block-types/ValidateCodeBlock.vue#lines-10

This way the maintenance will be less painful because we don't need to add `isEditable` for future block config elements. 

Here is a overview how to disable it
- change the element having `tree-sidebar` class into `<fieldset>`
- put conditional disabled there
Screen Shot 2021-02-23 at 1 09 39 PM

Comment (safydy-r) [reply src/components/common/AudioLibrarySearchField.vue:6]: But pls get confirmation from @bzabos first if he is good with this suggestion

File: src/components/common/AudioLibrarySearchField.vue
Comment (poojahpatil90) [reply src/components/common/AudioLibrarySearchField.vue:72]: @safydy-r  Sorry about the indentation fixes. My IDE was setup to auto fix the ESLint issues, which I did not realize until a later point and it was difficult to revert back those changes. 
One solution for now is to hide white-space changes from the settings button above(as suggested by Brett)

File: src/components/common/AudioLibrarySearchField.vue
Comment (poojahpatil90) [reply src/components/common/AudioLibrarySearchField.vue:6]: @safydy-r  Thanks for your suggestions. I will try out fieldset and update the PR.

**Comment by poojahpatil90** [comment]
I have removed the semicolon changes on this PR. Please choose 'hide whitespace changes' available on the settings button.  Although there are some other changes that I could not prevent, it will be much better to go over my changes now.

File: stories/Toolbar.stories.ts
Comment (safydy-r) [inline stories/Toolbar.stories.ts:58]: @poojahpatil90 , let's pass boolean type as defined in IBuilderState

File: src/store/builder/index.ts
Comment (safydy-r) [inline src/store/builder/index.ts:57]: @poojahpatil90 , let's put `true` by default. Otherwise all stories for blocks in storybook will have **disabled** block config

File: tests/unit/__snapshots__/storybook.spec.ts.snap
Comment (safydy-r) [inline tests/unit/__snapshots__/storybook.spec.ts.snap:145]: @poojahpatil90 , I'm happy you quickly learnt how to handle this, good job

File: stories/Toolbar.stories.ts
Comment (poojahpatil90) [reply stories/Toolbar.stories.ts:58]: Yes, I missed that. I will make the change.

File: src/store/builder/index.ts
Comment (poojahpatil90) [reply src/store/builder/index.ts:57]: Sure, I will make this change.

File: tests/unit/__snapshots__/storybook.spec.ts.snap
Comment (poojahpatil90) [reply tests/unit/__snapshots__/storybook.spec.ts.snap:145]: Thanks!  My knowledge on storybook is still limited. I am looking forward to learning and playing around with it more.

**Review by safydy-r — APPROVED** [review]

File: src/components/common/AudioLibrarySearchField.vue
Comment (bzabos) [reply src/components/common/AudioLibrarySearchField.vue]: Good one, I've been pushing for us to skip semi-colons!

**Review by bzabos — APPROVED** [review]
Thanks @safydy for the recommendation to use fieldsets! That made me smile :) To note: We'll need to explicitly pass `isEditable` for some non-native html components, but overall I favour this approach as well.

Looks good @poojahpatil90, thanks!

File: src/components/interaction-designer/flow-editors/ExitBlockEditorToggle.vue
Comment (poojahpatil90) [reply src/components/interaction-designer/flow-editors/ExitBlockEditorToggle.vue:3]: Added a note to handle this part as part of VMO-3546(for improving the view mode)

File: src/components/common/PlainDraggable.vue
Comment (poojahpatil90) [reply src/components/common/PlainDraggable.vue:72]: We could standardize this behavior if it makes sense :)

File: src/store/builder/index.ts
Comment (poojahpatil90) [reply src/store/builder/index.ts:57]: Done!

File: src/views/InteractionDesigner.vue
Comment (poojahpatil90) [reply src/views/InteractionDesigner.vue:205]: You are right, it is not ideal to use watchers. I do not like using them either, but in this case nothing else worked for me(at the time I was coding). I will take some time to explore other ways to catch changes on props effectively. I have added a note to tackle this as part of VMO-3546.

Comment (bzabos) [reply src/views/InteractionDesigner.vue:205]: Thanks! And you can always reach out to me if you want to brainstorm ways to avoid watchers :)

**Pull request merged**
